src/lib/interceptor.ts

```diff
@@ -240,6 +246,54 @@ export class ResponseParsingInterceptor {
   }
 }
 
+export class PreventMutationsInterceptor {
+  private requestInterceptor: grpc.Requester;
+  private blankInterceptor: grpc.Requester;
+
+  constructor() {
+    this.requestInterceptor = this.buildRequester();
+    this.blankInterceptor = buildBlankInterceptor();
+  }
+
+  public intercept(options: grpc.CallOptions, nextCall: NextCall): grpc.InterceptingCall {
+    let isMutationRequest = true;
+
+    if(options?.method_definition?.path) {
+      const { path } = options.method_definition;
+      for(const prefix of NON_MUTABLE_METHOD_PREFIXES) {
+        if(path.includes(prefix)) {
+          isMutationRequest = false;
+          break;
+        }
+      }
+    }
+
+    if(isMutationRequest) {
```

Comment:
If you use .includes or index of above, you could return there rather than checking again here